scripts/base/frameworks/packet-filter/main.zeek

```diff
@@ -48,6 +48,9 @@ export {
 
 		## Indicate if the filter was applied successfully.
 		success: bool  &log &default=T;
+
+		## A string reason why the filter failed to be created/installed.
+		reason: string &log &optional;
```

Comment:
One comparable field is in dpd.log saying `failure_reason` indicating it should hold error/failure information. Would that be worth adding to make it more descriptive?

The naming we use for this field varies across layers (pcap_geterr/errstr/my_error, state_message and reason up here), but that might just me getting confused. I understand we specifically don't treat the message as error everywhere.

Reply:
> One comparable field is in dpd.log saying failure_reason indicating it should hold error/failure information. Would that be worth adding to make it more descriptive?

Sure.

> The naming we use for this field varies across layers (pcap_geterr/errstr/my_error, state_message and reason up here), but that might just me getting confused. I understand we specifically don't treat the message as error everywhere.

I agree, but it's mostly just because those are the names of the things that are stored in the variables, such as `errstr` because we're calling `pcap_geterr`. Using `state` made sense to me because it's not really an error but more just the current state of the filter compilation process. `failure_reason` would make sense because it's only printed out on a failure (or well, a failure or a warning).